libnavigation-core/src/test/java/com/mapbox/navigation/core/trip/session/MapboxTripSessionTest.kt

```diff
@@ -1064,9 +1069,25 @@ class MapboxTripSessionTest {
         assertEquals(tripSession.routes, routes)
     }
 
+    @Test
+    fun `local routes reference updated after alternatives set`() = runBlockingTest {
+        coEvery { navigator.setAlternativeRoutes(any()) } coAnswers {
+            delay(100)
+        }
+
+        val routes = routes + mockk<NavigationRoute>()
+        tripSession.setRoutes(routes, legIndex, updateReason)
+
+        assertTrue(tripSession.routes.isEmpty())
+
+        coroutineRule.testDispatcher.advanceTimeBy(200)
+
+        assertEquals(tripSession.routes, routes)
+    }
+
```

Comment:
Native Navigator publishes updates as soon as primary route it set. It would be strange to have an empty routes on status update. Anyway, let's discuss it after we get [confirmation](https://github.com/mapbox/mapbox-navigation-android/pull/5587/files#r834014423)

Reply:
> as soon as primary route is set

We are setting a route in a coroutine that's scheduled to the back of the message queue, and await for the setting job to finish before `tripSession.routes` is updated to ensure that Nav Native actually already has the routes. Because of that, the routes will still remain empty right after `tripSession.setRoutes`, it's an async task.

Reply:
Sorry, the call is not set back to the message queue, but it's async nonetheless - we're waiting for NN to return before proceeding.